src/util/address.rs

```diff
@@ -150,6 +332,30 @@ pub enum AddressType {
     P2tr,
 }
 
+impl AddressType {
+    /// Extracts known [`AddressType`] from [`AddressInfo`], if any
+    #[inline]
+    pub fn with(address_info: AddressInfo) -> Option<AddressType> {
```

Comment:
I think `from_address_info` is the usual naming convention?

Reply:
the usual one is `TryFrom` :)

Reply:
Yes, `TryFrom` is better for conversions returning `Result`.

Reply:
+1 for `from_address_info` . I would find it more intuitive

Reply:
Done